lib/file-key-info.js

```diff
@@ -1,17 +1,16 @@
-var StringKeyInfo = require("./string-key-info"),
-  fs = require("fs");
+const fs = require("fs");
+const KeyInfo = require("./key-info");
 
 /**
- * A key info provider implementation
+ * A key info provider implementation for certificates in files
  *
  * @param {string} file path to public certificate
  */
-function FileKeyInfo(file) {
-  var key = fs.readFileSync(file);
-  StringKeyInfo.apply(this, [key]);
+class FileKeyInfo extends KeyInfo {
+  constructor(file) {
+    const cert = fs.readFileSync(file);
```

Comment:
The content of the X509Certificate element shouldn't have the `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` tags.
So it should either be stated that you expect a base64 encoded DER as input, or remove the tags if this is a PEM certificate.
Also, if it is a PEM file, it can contain multiple certificates, so you should parse them and put them in an array.

Reply:
Do you think a comment to that effect would be sufficient at this time?

Reply:
> Do you think a comment to that effect would be sufficient at this time?

Honestly, no. You should really modify the function to read a PEM chain from a file.
There is no point in this class otherwise.

The existing FileKeyInfo was meant to work with PEM files as well. From the docs:  

> A basic key info provider implementation using fs.readFileSync(file), is constructed using new FileKeyInfo([file]) where:
>     file - a path to a pem encoded certificate

Most people are used to working with PEM files. Nobody has a base64 encoded DER certificate file, and even if you make it, you can only put 1 certificate in there.
You added the functionality for the certificate to be an array of certificates, so this would be in line with that change.

I suggest you take a PEM file as input, and parse the certificate(s) inside into an array.

Reply:
Please have a look at the changes I've made and see if they address your concerns.